### PR TITLE
backend-app-api: deprecate plugin path configuration and fix /api/* 404

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

I think it's time for us to remove the configurability of backend paths. There is arguably extremely little long-term value in this configuration option, especially since you can still use the `indexPath` option of the root http router to lift up individual plugin APIs to top-level. This will of course impact anyone that is currently using the `getPath` option to configure path, which is why I'm suggesting we just deprecate it at this point.

Along with the deprecation of `getPath` this also adds a index router bypass for `/api/*` requests on the root http router, fixing the issue where you'd be served `index.html` on unknown plugin routes when running the `app-backend`.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/backend-system/core-services/root-http-router.md [hunk 1]
```diff
@@ -74,3 +76,5 @@ backend.add(
   }),
 );
 ```
+
+Note that requests towards `/api/*` will always fall through the `routes` if there is no matching plugin route, and never be caught by the configured `indexPath`.
```
Comment (freben) [inline docs/backend-system/core-services/root-http-router.md:80]: Maybe reword "fall through the routes", that made me stop for a sec and wonder what you meant

File: packages/backend-app-api/src/services/implementations/rootHttpRouter/DefaultRootHttpRouter.ts [hunk 2]
```diff
@@ -66,6 +66,12 @@ export class DefaultRootHttpRouter implements RootHttpRouterService {
   private constructor(indexPath?: string) {
     this.#indexPath = indexPath;
     this.#router.use(this.#namedRoutes);
+
+    // Any request with a /api/ prefix will skip the index router, even if no named router matches
+    this.#router.use('/api/', (_req, _res, next) => {
+      next('router');
```
Comment (freben) [inline packages/backend-app-api/src/services/implementations/rootHttpRouter/DefaultRootHttpRouter.ts:72]: Is this safe to do if `this.#indexPath` is false and there is no next router? :D

**Review by freben — APPROVED** [review]

File: packages/backend-app-api/src/services/implementations/rootHttpRouter/DefaultRootHttpRouter.ts (hunk 2 above)
Comment (Rugvip) [reply packages/backend-app-api/src/services/implementations/rootHttpRouter/DefaultRootHttpRouter.ts:72]: Yep, since we're exiting out from the `this.#router` it doesn't matter whether we have a handler for the index router registered afterwards or not. I added a test for that case though 👍
